pkg/apis/serving/v1alpha1/revision_types.go

```diff
@@ -337,3 +342,94 @@ func (rs *RevisionStatus) GetConditions() duckv1alpha1.Conditions {
 func (rs *RevisionStatus) SetConditions(conditions duckv1alpha1.Conditions) {
 	rs.Conditions = conditions
 }
+
+const (
+	AnnotationParseErrorTypeMissing = "Empty"
+	AnnotationParseErrorTypeInvalid = "Invalid"
+)
+
+// +k8s:deepcopy-gen=false
+type AnnotationParseError struct {
+	Type  string
+	Value string
+	Err   error
+}
+
+// +k8s:deepcopy-gen=false
+type LastPinnedParseError AnnotationParseError
+
+func (e LastPinnedParseError) Error() string {
+	return fmt.Sprintf("%v lastPinned value: \"%v\"", e.Type, e.Value)
+}
+
+// +k8s:deepcopy-gen=false
+type ConfigurationGenerationParseError AnnotationParseError
```

Comment:
Do these need to be public?

Reply:
LatPinnedParseError needs to be, but this one (configuration) does not.